### PR TITLE
WT-10071 Add nonstandalone evergreen tests

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by lukech** [comment]
@quchenhao I mainly reviewed the evergreen.yml changes. A few general questions/comments:

- I can see you are extending the scope of the ticket to introduce non-standalone tests against a few different testing frameworks (Python, CPP, stress). While having more coverage is an advantage, I think we need to be aware of the additional building resources coming with it, to make sure we get the maximal value from those new tests. Can we add one or a few non-standalone tests/tasks for each of the testing frameworks at the current stage, and observe the failure capture efficiency, before introducing more thorough coverage? 
- Is it worth identifying a few tasks that can be added to the pull request testing scope? (adding the `pull_request` tag)
- It's good to apply the tag/selector to make the code neater. I still found there are quite some code duplications that could potentially be avoided.

File: test/evergreen.yml [hunk 1]
```diff
@@ -961,6 +979,16 @@ tasks:
           CLANG_C_VERSION: -DCLANG_C_VERSION=8
           CLANG_CXX_VERSION: -DCLANG_CXX_VERSION=8
 
+  # Base compile for nonstandalone build
+  - name: compile-nonstandalone
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          NONSTANDALONE: -DWT_STANDALONE_BUILD=0
```
Comment (lukech) [inline test/evergreen.yml:988]: It feels worthwhile to retrieve this section of code into a separate function, as it's called multiple times in the file.

Comment (quchenhao) [reply test/evergreen.yml:988]: It is already a function. Not sure how would you do it.

File: test/evergreen.yml [hunk 2]
```diff
@@ -1427,6 +1463,149 @@ tasks:
...
+    tags: ["cppsuite-stress-test-nonstandalone"]
+    commands:
+      - func: "fetch artifacts"
+        vars:
+          dependent_task: compile-nonstandalone
+      - func: "cppsuite test"
+        vars:
+          test_config_filename: configs/search_near_03_stress.txt
+          test_name: search_near_03
+
```
Comment (lukech) [inline test/evergreen.yml:1608]: Do we really need that many CPP tests to be covered? Can we embed the non-standalone tests into the corresponding standalone tests (e.g. adding another "compile non-standalone" function and running the test again)?

File: test/evergreen.yml [hunk 3]
```diff
@@ -4329,6 +4770,12 @@ buildvariants:
     - name: ".stress-test-sanitizer-3"
     - name: ".stress-test-sanitizer-4"
     - name: format-abort-recovery-stress-test
+    - name: ".stress-test-1-nonstandalone"
+    - name: ".stress-test-2-nonstandalone"
+    - name: ".stress-test-3-nonstandalone"
+    - name: ".stress-test-4-nonstandalone"
+    - name: ".stress-test-no-barrier-nonstandalone"
+    - name: format-abort-recovery-stress-test-nonstandalone
```
Comment (lukech) [inline test/evergreen.yml:4778]: It feels a bit overkill to have this many non-standalone stress tests running on the ASAN builder. Can we reduce it to 1 or 2?

File: test/evergreen.yml [hunk 4]
```diff
@@ -4358,6 +4805,12 @@ buildvariants:
     - name: ".stress-test-4"
     - name: ".stress-test-no-barrier"
     - name: format-abort-recovery-stress-test
+    - name: ".stress-test-1-nonstandalone"
+    - name: ".stress-test-2-nonstandalone"
+    - name: ".stress-test-3-nonstandalone"
+    - name: ".stress-test-4-nonstandalone"
+    - name: ".stress-test-no-barrier-nonstandalone"
+    - name: format-abort-recovery-stress-test-nonstandalone
```
Comment (lukech) [inline test/evergreen.yml:4813]: Ditto

File: test/evergreen.yml (hunk 2 above)
Comment (quchenhao) [reply test/evergreen.yml:1608]: I think stress tests need to be covered. Will embed the tests into standalone alone tests save resources? I think the workload to run is the same. Better to keep them separate to be clean. We will delete these tests once the data format change is released to mongodb

File: test/evergreen.yml [hunk 5]
```diff
@@ -4160,20 +4606,15 @@ buildvariants:
...
-    - name: data-validation-stress-test-checkpoint-fp-hs-insert-s6
-    - name: data-validation-stress-test-checkpoint-fp-hs-insert-s7
-    - name: data-validation-stress-test-checkpoint-no-timestamp
+    - name: ".data-validation-stress-test"
     - name: unittest-test
     - name: tiered-storage-extensions-test
+    - name: compile-nonstandalone
+    - name: make-check-nonstandalone
+    - name: unit-test-nonstandalone
+    - name: unit-test-long-nonstandalone
```
Comment (lukech) [inline test/evergreen.yml:4615]: I'd prefer not to introduce the full set of make check + unit test + unit test long tasks.

**Review by lukech — APPROVED** [review]
I'm going to approve the PR to let it merge and enable the new tests to run and uncover issues. There are a few open threads, some are related to the added test cycles and balance between standalone and non-standalone modes (that we can file a follow-on ticket to discuss), and the others are related to evergreen.yml coding-style/deduplication which is not critical. 

Please run a patch build to test the newly added tasks before merge, and check artifacts downloading details to avoid running into the same issue that caused the previous revert.

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**